### PR TITLE
Add USB video kernel support

### PR DESCRIPTION
**Pull request by DaAwesomeP**
This looks excessive but this seems to be required given the completely different requirements for the different webcams I tested with (I was able to get them all to appear in `/dev` and have their drivers loaded). Since support is not quite baked into everything else yet, I was not able to test beyond ensuring they show up and the right modules are loaded and that `dmesg` doesn't complain.

I saw some of the other boards have some subset of these parameters, but I did not modify them since I have no way to test them and am unsure of any other platform specifics.

This includes webcams and most simple USB capture cards, so it should solve #491 at least for `intel_nuc`.

### CODE REVIEW

**Comment by DaAwesomeP** [comment]
@pvizeli Looking at past pulls it looks like maybe this should go in ` operating-system/buildroot-external/kernel/device-support.config `? Please let me know if I should make that change.

Assuming there are no caveats with any of these configs in the other platforms, I suppose the duplicate configs lines could be removed from other board-specific configs if this is added to `device-support.config`?

**Comment by DaAwesomeP** [comment]
OK, I just installed the latest (unreleased) version of the motion eye addon that adds the video option to the addon config, and I can confirm that this works! Cameras can be viewed and controlled properly.

**Comment by pvizeli** [comment]
What will be the +size of all the modules? Can you group those options and explain a bit more about why that module/group is required and why you listen in there.

> device-support.config
Yes, devices they are plug&play like USB stuff and can be used also by other hardware, need to listen on this file. Just hardware they are only recommended for this kind of image, are in the board kernel config.

**Comment by DaAwesomeP** [comment]
> What will be the +size of all the modules?
My build went up by about two MB.

> Can you group those options and explain a bit more about why that module/group is required and why you listen in there.

Yes, commit imminent!

> Yes, devices they are plug&play like USB stuff and can be used also by other hardware, need to listen on this file. Just hardware they are only recommended for this kind of image, are in the board kernel config.

Cool, I will move it over and sort through any duplicates in other files. Yes, that makes sense (i.e. the RPI video stuff should only be in those configs)

**Comment by DaAwesomeP** [comment]
OK, without the video stuff intel_nuc is 219M. With the video configs it is 221M.

I have loaded this updated commit on my device and I can confirm it works.

File: buildroot-external/kernel/device-support.config [hunk 1]
```diff
@@ -24,3 +24,171 @@ CONFIG_BT_RTL=m
...
+CONFIG_VIDEOBUF2_DMA_CONTIG=m
+CONFIG_VIDEOBUF2_VMALLOC=m
+CONFIG_VIDEOBUF2_DMA_SG=m
+CONFIG_VIDEOBUF2_DVB=m
+CONFIG_DVB_B2C2_FLEXCOP=m
+CONFIG_VIDEO_SAA7146=m
+CONFIG_VIDEO_SAA7146_VV=m
+CONFIG_SMS_SIANO_MDTV=m
+CONFIG_SMS_SIANO_RC=y
+CONFIG_VIDEO_V4L2_TPG=m
```
Comment (pvizeli) [inline buildroot-external/kernel/device-support.config]: Move this to the common devices

File: buildroot-external/kernel/device-support.config [hunk 2]
```diff
@@ -24,3 +24,171 @@ CONFIG_BT_RTL=m
...
+CONFIG_MEDIA_DIGITAL_TV_SUPPORT=y
+CONFIG_MEDIA_USB_SUPPORT=y
+CONFIG_MEDIA_PCI_SUPPORT=y
+CONFIG_VIDEO_DEV=m
+CONFIG_VIDEO_V4L2_SUBDEV_API=y
+CONFIG_VIDEO_V4L2=m
+CONFIG_VIDEOBUF_GEN=m
+CONFIG_VIDEOBUF_DMA_SG=m
+CONFIG_VIDEOBUF_VMALLOC=m
+CONFIG_V4L_PLATFORM_DRIVERS=y
```
Comment (pvizeli) [inline buildroot-external/kernel/device-support.config]: This part also to the common devices

**Comment by DaAwesomeP** [comment]
> Move this section to the common devices
> This section also to the common devices

OK, I have grouped the first and last sections together

> Can we remove that # Analog/digital TV/input USB devices ? We have no use case for this and also no add-on which brings tv support or something else. In the end, I need to maintain that until the end of my life and is not funny to fix the issue or add a kernel patch for things where no one is used.

Yes, good point—there is no need for tuners. This list originally came from the Odroid config that I trimmed down a little.

I removed all of the DVB devices (TV tuners) and went through the rest of the non-webcam device configs to ensure they are all used for direct capture card modules (e.g. composite/component/S-video input only). These were originally labeled as "TV" in the Linux kernel comment since the previous most common use was to plug them into the output of a cable box and they are often found inside of devices that also have a demodulator.

This is what is being used in #491. These analog non-TV, tunerless capture devices are now most often used to connect to an analog security camera or the multi-view output of an NVR for remote monitoring/streaming.

I have not tested this commit and won't have a chance to until the end of the week, but I am 90% sure it should work fine since I only removed devices.

**Comment by pvizeli** [comment]
ok. Thanks for contribution

**Comment by DaAwesomeP** [comment]
Thank you for maintaining such an awesome project!

**Pull request merged**